sway/container.c

```diff
@@ -26,14 +26,11 @@ static void free_swayc(swayc_t *cont) {
 	if (!ASSERT_NONNULL(cont)) {
 		return;
 	}
-	// TODO does not properly handle containers with children,
-	// TODO but functions that call this usually check for that
 	if (cont->children) {
-		if (cont->children->length) {
-			int i;
-			for (i = 0; i < cont->children->length; ++i) {
-				free_swayc(cont->children->items[i]);
-			}
+		// remove children until there are no more, free_swayc calls
+		// remove_child, which removes child from this container
+		while (cont->children->length) {
+			free_swayc(cont->children->items[0]);
```

Comment:
forgot to add this.
as the comment says, free_swayc removes the child from parent, so we continue until length is 0.
before it would i=0 len=5 -> i=1 len=4 -> i=2 len=3 -> i=3 len=2 break; and not free 3 or 4